model/riscv_vmem_sv32.sail

```diff
@@ -74,7 +74,7 @@
 
 function to_phys_addr(a : paddr32) -> xlenbits = a[31..0]
 
-val walk32 : (vaddr32, AccessType(ext_access_type), Privilege, bool, bool, paddr32, nat, bool, ext_ptw) -> PTW_Result(paddr32, SV32_PTE) effect {rmem, rmemt, rreg, escape}
+val walk32 : (vaddr32, AccessType(ext_access_type), Privilege, bool, bool, paddr32, nat, bool, ext_ptw) -> PTW_Result(paddr32, SV32_PTE) effect {rmem, rmemt, rreg, wreg, escape}
```

Comment:
This is dodgy. Architecturally page table walks do not write to registers. It's only because as a side-effect they stash state for RVFI to later report.

Reply:
Similar is true for the walk39 and walk48 in that case

Reply:
Yes, I only commented once on it

Reply:
Hmm, that's a pain: is there a way to guard the effect with an ifdef of whether rvfi is enabled?

Reply:
Although it looks to me like the first function we call `mem_read_priv` has `wreg`, so (assuming you have to inherit the effects of callees), there is currently a bug here.

Reply:
Isn't there a danger that `$ifdef`-ed signatures spread throughout the code base?